File: LLM-Translations/GPT-4/codenet/C++/Java/s343280137.java
import java.util.Scanner;

public class s343280137 {

    private static final long MOD = 1000000007;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        long n = scanner.nextLong();
        long c = scanner.nextLong();

        long[] x = new long[(int) (n + 1)];
        long[] v = new long[(int) (n + 1)];

        for (int i = 1; i < n + 1; i++) {
            x[i] = scanner.nextLong();
            v[i] = scanner.nextLong();
        }

        long[] leftTotal = new long[(int) (n + 1)];
        long[] maxLeftTotal = new long[(int) (n + 1)];
        for (int i = 1; i < n + 1; i++) {
            leftTotal[i] = leftTotal[i - 1] + v[i];
        }

        for (int i = 1; i < n + 1; i++) {
            maxLeftTotal[i] = Math.max(maxLeftTotal[i - 1], leftTotal[i] - x[i]);
        }

        long[] rightTotal = new long[(int) (n + 1)];
        long[] maxRightTotal = new long[(int) (n + 1)];
        for (int i = 1; i < n + 1; i++) {
            rightTotal[i] = rightTotal[i - 1] + v[(int) (n - i + 1)];
        }

        for (int i = 1; i < n + 1; i++) {
            maxRightTotal[i] = Math.max(maxRightTotal[i - 1], rightTotal[i] - (c - x[(int) (n + 1 - i)]));
        }

        long ans = 0;

        for (int i = 1; i < n + 1; i++) {
            ans = Math.max(ans, leftTotal[i] - x[i]);
            ans = Math.max(ans, leftTotal[i] - 2 * x[i] + maxRightTotal[(int) (n - i)]);
            ans = Math.max(ans, rightTotal[i] - (c - x[(int) (n + 1 - i)]));
            ans = Math.max(ans, rightTotal[i] - 2 * (c - x[(int) (n + 1 - i)]) + maxLeftTotal[(int) (n - i)]);
        }

        System.out.println(ans);
    }
}
// 

File: LLM-Translations/GPT-4/codenet/C++/Go/s343280137.go

package main

import (
	"fmt"
	"math"
)

const MOD int64 = 1000000007

func main() {
	var n, c int64
	fmt.Scan(&n, &c)

	x := make([]int64, n+1)
	v := make([]int64, n+1)

	for i := 1; i <= int(n); i++ {
		fmt.Scan(&x[i], &v[i])
	}

	lefttotal := make([]int64, n+1)
	maxlefttotal := make([]int64, n+1)
	for i := 1; i <= int(n); i++ {
		lefttotal[i] = lefttotal[i-1] + v[i]
	}

	for i := 1; i <= int(n); i++ {
		maxlefttotal[i] = int64(math.Max(float64(maxlefttotal[i-1]), float64(lefttotal[i]-x[i])))
	}

	righttotal := make([]int64, n+1)
	maxrighttotal := make([]int64, n+1)
	for i := 1; i <= int(n); i++ {
		righttotal[i] = righttotal[i-1] + v[int(n)-i+1]
	}

	for i := 1; i <= int(n); i++ {
		maxrighttotal[i] = int64(math.Max(float64(maxrighttotal[i-1]), float64(righttotal[i]-(c-x[int(n)+1-i]))))
	}

	var ans int64 = 0

	for i := 1; i <= int(n); i++ {
		ans = int64(math.Max(float64(ans), float64(lefttotal[i]-x[i])))
		ans = int64(math.Max(float64(ans), float64(lefttotal[i]-2*x[i]+maxrighttotal[int(n)-i])))
		ans = int64(math.Max(float64(ans), float64(righttotal[i]-(c-x[int(n)+1-i]))))
		ans = int64(math.Max(float64(ans), float64(righttotal[i]-2*(c-x[int(n)+1-i])+maxlefttotal[int(n)-i])))
	}

	fmt.Println(ans)

}

// 
